pkg/channel/event_receiver.go

```diff
@@ -218,6 +245,29 @@ func (r *EventReceiver) ServeHTTP(response nethttp.ResponseWriter, request *neth
 		return
 	}
 
+	/// Here we do the OIDC audience verification
+	features := feature.FromContext(ctx)
+	if features.IsOIDCAuthentication() {
+		r.logger.Debug("OIDC authentication is enabled")
+
+		token := auth.GetJWTFromHeader(request.Header)
+		if token == "" {
+			r.logger.Warn(fmt.Sprintf("No JWT in %s header provided while feature %s is enabled", auth.AuthHeaderKey, feature.OIDCAuthentication))
+			response.WriteHeader(nethttp.StatusUnauthorized)
+			return
+		}
+
+		if _, err := r.tokenVerifier.VerifyJWT(ctx, token, r.audience); err != nil {
+			r.logger.Warn("no valid JWT provided", zap.Error(err))
+			response.WriteHeader(nethttp.StatusUnauthorized)
+			return
+		}
+
+		r.logger.Debug("Request contained a valid JWT. Continuing...")
+	} else {
+		r.logger.Debug("OIDC authentication is disabled")
+	}
+
```

Comment:
Could we refactor this into a method in the `auth` package? I.e. something similar to https://github.com/knative/eventing/issues/7377?

I know we need essentially the same logic when we do this for mt-broker-filter, and maybe elsewhere too?

Reply:
As we have this probably in multiple places (ingress, channel receiver and filter), I am fine too with creating an issue for this and solve this independently from this PR. But whatever works best here.

Reply:
Thanks for creating the issue @Leo6Leo: https://github.com/knative/eventing/issues/7487